src/operators/suffix.ts

```diff
@@ -139,6 +139,23 @@ export class SuffixOperator implements Operator {
     }
   }
 
+  /**
+   * Check to see if a prop already contains a suffix
+   * @param prop The prop to check
+   * @return If the prop has a suffix on it already
+   */
+  static isAlreadySuffixed(prop: string) : boolean {
+    if (prop === undefined || prop === null) {
+      return false;
+    }
+    let matches = false;
+    const suffixes:Array<Suffixes> = Object.values(Suffixes);
```

Comment:
It seems like we could move this outside the function scope and call `Object.values` once instead of once for every property we evaluate. I'm not sure if we'll run into friction with the current `static` modifier on this function, though.

If we're only calling `Object.values` once, we could also consider reducing these values to an object with keys matching known type suffixes. That'd give us way to check a substring against object keys. I'm not sure if performance is a concern here or which route would be more performant, but capturing the last underscore index and checking any property suffix against an object with keys containing known suffixes is how the FullStory Segment cloud mode integration checks for known type suffixes. This also allows you to short-circuit the O(n) `endsWith` check against all known type suffixes if there isn't an underscore as @van-fs was suggesting. Examples from our open source Segment cloud mode integration:

- [Type suffix validators containing underscore-less suffixes as keys](https://github.com/segmentio/action-destinations/blob/3356398b24c12eda10495b75e73afa72d6c1968f/packages/destination-actions/src/destinations/fullstory/vars.ts#L37-L52)
- [Check for known type suffixes](https://github.com/segmentio/action-destinations/blob/3356398b24c12eda10495b75e73afa72d6c1968f/packages/destination-actions/src/destinations/fullstory/vars.ts#L98-L100)

Feel free to take or leave any of these ideas, but extracting the `Object.values` call may be worth considering if that change doesn't introduce too much friction.

Reply:
I'll move the Object.keys into a variable at the same level as Suffixes so we don't have to do it over and over.  I don't think it is a huge performance gain, but it is an easy fix.